### ISSUE TITLE
Rust errors when enabling `dns-over-https` feature

### ISSUE THREAD

**Issue by williamdes** [bluejekyll/trust-dns#1648]
**Describe the bug**

It's possible that part of the issue would need `rustls` to be added as an optional dependency like `h2` is 🤷🏻‍♂️ 
```
   Compiling trust-dns-client v0.21.0 (/workspace/crates/client)
error[E0432]: unresolved import `rustls`
  --> crates/client/src/https_client_connection.rs:14:5
   |
14 | use rustls::ClientConfig;
   |     ^^^^^^ help: a similar path exists: `trust_dns_proto::rustls`

   Compiling trust-dns-resolver v0.21.0 (/workspace/crates/resolver)
   Compiling trust-dns-native-tls v0.21.0 (/workspace/crates/native-tls)
   Compiling trust-dns-https v0.21.0 (/workspace/crates/https)
   Compiling trust-dns-rustls v0.21.0 (/workspace/crates/rustls)
For more information about this error, try `rustc --explain E0432`.
error: could not compile `trust-dns-client` due to previous error
warning: build failed, waiting for other jobs to finish...
error[E0432]: unresolved import `crate::tls`
 --> crates/resolver/src/https.rs:4:12
  |
4 | use crate::tls::CLIENT_CONFIG;
  |            ^^^ could not find `tls` in the crate root

error[E0432]: unresolved import `crate::config::TlsClientConfig`
  --> crates/resolver/src/https.rs:10:5
   |
10 | use crate::config::TlsClientConfig;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `TlsClientConfig` in `config`

error[E0425]: cannot find value `client_config` in this scope
   --> crates/resolver/src/name_server/connection_provider.rs:195:21
    |
195 |                     client_config,
    |                     ^^^^^^^^^^^^^ not found in this scope

Some errors have detailed explanations: E0425, E0432.
For more information about an error, try `rustc --explain E0425`.

```

**To Reproduce**
Clone https://github.com/trust-dns/docker
Edit `alpine/Dockerfile` to add the feature `dns-over-https` on line 20

**Expected behavior**
Build

**System:**
 - OS:Alpine 3.14
 - rustc version: 1.59
 
Same as for #1647

**Version:**
Version: 0.21.0

**Additional context**
Feature: `dns-over-https`

**Comment by bluejekyll** [comment]
This is described here on the readme: https://github.com/bluejekyll/trust-dns#dns-over-tls-and-dns-over-https

basically, the feature you want is `dns-over-https-rustls`

**Comment by bluejekyll** [comment]
Is this resolved along with #1647 now?

**Comment by williamdes** [comment]
> Is this resolved along with #1647 now?

Yes, thank you 🚀 
https://github.com/trust-dns/docker/actions/runs/1909126348
